Player controls - NETWORKED

nfetissow opened the pull request:
Sorry for ugly and huge PR, next ones will be smaller, I promise.

Commit history speaks for itself: 
1. Added player control and simple QTE engine, that is triggered by pressing the same button 3 times in a row.
2. Changed the Networking part a bit, so that it is easier to use. The biggest issue might be is that I allow "Empty" as a message type - not all NetworkBehaviours that I added need custom messages. @momower1 , are you fine with this change?

@maxwerhahn idea for Ship-Environment interaction is as follows:
1. Get Ship object from Gamemaster.
2. Get "Manipulator"-type object from the Ship. Currently, there is "ForceManipulator", that enables its user to apply force on a Ship for a set amount of time (indefinitely too).

Idea is that the Ship (and players if necessary) will have those Manipulators attached to them, and then you change Ship (or player) by calling its methods. On the server it just does it right away, on the client it sends messages to the server with proper parameters.

Let me know what you think, and also if you need some other Manipulators besides ForceManipulator for your work, I'll quickly add dummies that'll be logging calls until I actually implement them.

To test what I have simply change scenes in the LobbyManager to ShipSceneServer and ShipSceneClient.

momower1 commented:
@nfetissow 
You should not allow Empty as a message type because then the messages will be received by all NetworkBehaviours with that type and mybe they can not handle this type of message. If you don't need a message type for each of your class then just create a new one which is shared for the classes that you implemented yourself. I just think it's a huge risk to allow the Empty message type.

nfetissow commented:
@momower1 I've added new type "NewMessages". The "Empty" will be throwing errors now, as we can assume that the developer forgot to set up the prefab properly. However, if you specify "NoMessages" and then go on and try to send custom messages anyway, and it doesn't work, then it's your own fault and you shouldn't do it. 
I think it shouldn't cause us problems down the way.

@maxwerhahn  I've added the AddForceAtPosition method, however it doesn't do anything new atm. But you can already use it like it exists. Also let me know whether the position will be supplied in world-space, or in local space of the ship.

maxwerhahn commented:
@nfetissow The position will be in world-space.
